handlers/handlers.go

```diff
@@ -55,12 +58,78 @@ func (d *CatalystAPIHandlersCollection) TranscodeSegment() httprouter.Handle {
 			errors.WriteHTTPBadRequest(w, "Invalid request payload", err)
 			return
 		}
-
 		callbackClient := clients.NewCallbackClient()
-		if err := callbackClient.SendTranscodeStatusError(transcodeRequest.CallbackUrl, "NYI - not yet implemented"); err != nil {
-			errors.WriteHTTPInternalServerError(w, "error send transcode error", err)
+		errorOutInternal := func(where string, err error) {
+			// Mist does not respond or handle returned error codes. We do this for correctness.
+			errors.WriteHTTPInternalServerError(w, where, err)
+			if err := callbackClient.SendSegmentTranscodeError(transcodeRequest.CallbackUrl, where, err.Error(), transcodeRequest.SourceFile); err != nil {
+				fmt.Printf("error send transcode error %v\n", err)
+				return
+			}
+		}
+
+		inputStream, renditionsStream := generateStreamNames()
+		if err = d.MistClient.AddStream(inputStream, transcodeRequest.SourceFile); err != nil {
+			errorOutInternal("error AddStream(inputStream)", err)
+			return
+		}
+
+		// Start MistProcLivepeer
+		configPayload, err := json.Marshal(configForSubprocess(&transcodeRequest, broadcasterPort, inputStream, renditionsStream))
+		if err != nil {
+			errorOutInternal("error ProcLivepeerConfig json encode", err)
+			return
+		}
+		fmt.Printf("configPayload %s\n", configPayload)
+		transcodeCommand := exec.Command(mistProcPath, "-")
+		stdinPipe, err := transcodeCommand.StdinPipe()
+		if err != nil {
+			errorOutInternal("error transcodeCommand.StdinPipe()", err)
+			return
+		}
+		commandOutputToLog(transcodeCommand, "coding")
+		sent, err := stdinPipe.Write(configPayload)
+		if err != nil {
+			errorOutInternal("error stdinPipe.Write()", err)
+			return
+		}
+		if sent != len(configPayload) {
+			errorOutInternal("error short write on stdinPipe.Write()", err)
+			return
+		}
+		err = stdinPipe.Close()
+		if err != nil {
+			errorOutInternal("error stdinPipe.Close()", err)
+			return
+		}
+		err = transcodeCommand.Start()
+		if err != nil {
+			errorOutInternal("error start transcodeCommand", err)
+			return
+		}
+
+		// TODO: remove when Mist code is updated <
+		// Starting SOURCE_PREFIX stream. Why we need to start it manually? MistProcLivepeer should start it??
+		if err := d.MistClient.PushStart(inputStream, "/opt/null.ts"); err != nil {
+			errorOutInternal("error PushStart(inputStream)", err)
+			return
+		}
```

Comment:
Waiting on @Thulinma changes to remove this push step.

Reply:
Ok, then should this PR be converted to draft and merged after the changes to @Thulinma ? I think we should try to avoid such `TODOs` in the `main` branch.